src/grammar/query.rs

```diff
@@ -172,14 +172,197 @@ pub(crate) fn parse_into_clause(p: &mut Parser, expect_into_clause: bool) {
 }
 
 fn parse_from_list(p: &mut Parser) {
+    let mut expect_join = false;
     safe_loop!(p, {
-        parse_ident(p, 1..1);
-        if !p.eat(T![,]) {
+        if !expect_join {
+            parse_ident(p, 1..1);
+        }
+        if [
+            T![join],
+            T!["("],
+            T![inner],
+            T![outer],
+            T![cross],
+            T![natural],
+            T![left],
+            T![right],
+            T![full],
```

Comment:
Could we store this in a variable? Just to be sure if we have to change it, we wouldn't forget it at the other location.